src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java

```diff
@@ -69,4 +69,6 @@
     Long countByAssessorIsNotNullAndParticipation_Exercise_CourseId(long courseId);
 
     Long countByAssessor_IdAndParticipation_Exercise_CourseId(long assessorId, long courseId);
+
+    List<Result> findAllByParticipation_Exercise_CourseId(Long courseId);
```

Comment:
Yes, these methods are a bit magic, they generate queries based only on their names.

The `_` indicates the traversal point between tables. You can omit it, but then the query would be ambiguous, and if in the future we add new properties to the data model it could fail.

Adding the `_` makes it super clear to Spring Data what we are referring to

https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repositories.query-methods.query-property-expressions

Reply:
thanks good to know

Reply:
@rpadovani Can you please add a JavaDoc comment to this method with the explanation of this conversation? Thanks

Reply:
@krusche I actually think this is not the right place, since it is a framework feature: maybe we should create a page on confluence with "Good to know" or something like that explaining some of the things we use that probably new students haven't seen yet, like this, streams, <fa-icon> and similar?